From 2a7e26e1141692893b0b8fb9c4898fa4b34f7def Mon Sep 17 00:00:00 2001
From: Loc Vu <loc.vu.zn@renesas.com>
Date: Fri, 19 Jan 2018 17:31:33 +0700
Subject: [PATCH 033/433] ARM: dts r8a7744: Add clocks Declare all core

clocks and DIV6 clocks, as well as all MSTP clocks
currently used by r8a7744 boards.

Signed-off-by: thongsyho <thong.ho.px@rvc.renesas.com>
Signed-off-by: Loc Vu <loc.vu.zn@renesas.com>
---
 arch/arm/boot/dts/r8a7744.dtsi | 38 +++++++++++++++++++++++++++++++++++++-
 1 file changed, 37 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/r8a7744.dtsi b/arch/arm/boot/dts/r8a7744.dtsi
index 312c9aa..9b97183 100644
--- a/arch/arm/boot/dts/r8a7744.dtsi
+++ b/arch/arm/boot/dts/r8a7744.dtsi
@@ -23,18 +23,21 @@
 		compatible = "fixed-clock";
 		#clock-cells = <0>;
 		clock-frequency = <0>;
+		clock-output-names = "audio_clk_a";
 	};
 
 	audio_clk_b: audio_clk_b {
 		compatible = "fixed-clock";
 		#clock-cells = <0>;
 		clock-frequency = <0>;
+		clock-output-names = "audio_clk_b";
 	};
 
 	audio_clk_c: audio_clk_c {
 		compatible = "fixed-clock";
 		#clock-cells = <0>;
 		clock-frequency = <0>;
+		clock-output-names = "audio_clk_c";
 	};
 
 	/* External CAN clock */
@@ -43,6 +46,8 @@
 		#clock-cells = <0>;
 		/* This value must be overridden by the board. */
 		clock-frequency = <0>;
+		clock-output-names = "can_clk";
+		status = "disabled";
 	};
 
 	cpus {
@@ -1606,7 +1611,22 @@
 				#clock-cells = <0>;
 				clock-output-names = "mmc0";
 			};
-
+			ssp_clk: ssp_clk@e6150248 {
+				compatible = "renesas,r8a7744-div6-clock",
+					     "renesas,cpg-div6-clock";
+				reg = <0 0xe6150248 0 4>;
+				clocks = <&pll1_div2_clk>;
+				#clock-cells = <0>;
+				clock-output-names = "ssp";
+			};
+			ssprs_clk: ssprs_clk@e615024c {
+				compatible = "renesas,r8a7744-div6-clock",
+					     "renesas,cpg-div6-clock";
+				reg = <0 0xe615024c 0 4>;
+				clocks = <&pll1_div2_clk>;
+				#clock-cells = <0>;
+				clock-output-names = "ssprs";
+			};
 			/* Fixed factor clocks */
 			pll1_div2_clk: pll1_div2_clk {
 				compatible = "fixed-factor-clock";
@@ -1648,6 +1668,14 @@
 				clock-mult = <1>;
 				clock-output-names = "hp";
 			};
+			i_clk: i_clk {
+				compatible = "fixed-factor-clock";
+				clocks = <&cpg_clocks R8A7744_CLK_PLL1>;
+				#clock-cells = <0>;
+				clock-div = <2>;
+				clock-mult = <1>;
+				clock-output-names = "i";
+			};
 			b_clk: b_clk {
 				compatible = "fixed-factor-clock";
 				clocks = <&cpg_clocks R8A7744_CLK_PLL1>;
@@ -1680,6 +1708,14 @@
 				clock-mult = <1>;
 				clock-output-names = "m2";
 			};
+			imp_clk: imp_clk {
+				compatible = "fixed-factor-clock";
+				clocks = <&cpg_clocks R8A7744_CLK_PLL1>;
+				#clock-cells = <0>;
+				clock-div = <4>;
+				clock-mult = <1>;
+				clock-output-names = "imp";
+			};
 			rclk_clk: rclk_clk {
 				compatible = "fixed-factor-clock";
 				clocks = <&cpg_clocks R8A7744_CLK_PLL1>;
-- 
2.7.4

